### PR TITLE
Improving testing infrastructure

### PR DESCRIPTION
**Pull request by nhibberd**
ambiata/mundane#74

### CODE REVIEW

File: saws-s3/src/main/scala/com/ambiata/saws/s3/S3.scala [hunk 1]
```diff
@@ -13,7 +13,7 @@ object S3 {
     S3Action(_.listBuckets.toList).onResult(_.prependErrorMessage(s"Could access the buckets list"))
 
   def isS3Accessible: S3Action[Unit] =
-    listBuckets.map(_ => ()).orElse(S3Action.fail("S3 is not accessible"))
+    listBuckets.void.orElse(S3Action.fail("S3 is not accessible").void)
```
Comment (markhibberd) [inline saws-s3/src/main/scala/com/ambiata/saws/s3/S3.scala]: Doesn't matter that much, but you can probably just void this once on the outside.

File: saws-testing/src/test/scala/com/ambiata/saws/testing/S3Spec.scala [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+import org.specs2.matcher.Parameters
+import org.specs2.ScalaCheck
+import org.specs2.Specification
+import org.specs2.specification.Fragments
+
+abstract class S3Spec extends Specification with ScalaCheck {
+  override def map(fs: =>Fragments) = section("aws") ^ super.map(fs)
+
+  override implicit def defaultParameters: Parameters =
+    new Parameters(minTestsOk = 5, workers = 3)
```
Comment (markhibberd) [inline saws-testing/src/test/scala/com/ambiata/saws/testing/S3Spec.scala]: I am a bit worried about this, how long do you think before someone extends this not realizing or copy-pasting from another test, could maybe make the min tests and argument, so you go like `extends S3Spec(5)` or similar in test case so we can at least see in reviews.

File: saws-testing/src/test/scala/com/ambiata/saws/testing/S3Spec.scala [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package com.ambiata.saws.testing
+
+import org.specs2.matcher.Parameters
+import org.specs2.ScalaCheck
+import org.specs2.Specification
+import org.specs2.specification.Fragments
+
+abstract class S3Spec extends Specification with ScalaCheck {
```
Comment (markhibberd) [inline saws-testing/src/test/scala/com/ambiata/saws/testing/S3Spec.scala]: Maybe just AwsSpec, doesn't seem specific given tag.

**Comment by markhibberd** [comment]
Looking pretty good I think, I will do a more detailed review of the test transformations when I get a chance for you.

File: saws-s3/src/main/scala/com/ambiata/saws/s3/S3Pattern.scala [hunk 4]
```diff
@@ -53,7 +53,12 @@ case class S3Pattern(bucket: String, unknown: String) {
     listS3.map(_.map(_.s3.key))
 
   def delete: S3Action[Unit] =
-    resolve.map(_.map(_.delete))
+    determine.flatMap({
+      case Some(\/-(v)) => v.delete
+      case Some(-\/(v)) => v.delete
+      case None         => S3Action.unit
+    })
+//    resolve.flatMap(_.traverse(_.delete)).void
```
Comment (charleso) [inline saws-s3/src/main/scala/com/ambiata/saws/s3/S3Pattern.scala]: Remove dead code /nit

**Pull request merged**